File: GuessMyNumberGame.py
#/usr/bin/env python3
# -*- coding: utf-8 -*-

import random

difficulty = {'easy':random.randint(1,10),'medium':random.randint(1,100),'hard':random.randint(1,1000),'very hard':random.randint(1,10000)}
print("Hello and welcome to Roman Menzel's Guess My Number Game")
print("1.) Easy")
print("2.) Medium")
print("3.) Hard")
print("4.) Very Hard")
print("Your choice: ")
choice = input()

if choice == "1":
	tries = 0
	while True:
		print("Enter your guess: ")
		g = input()
		guess = int(g)
		if guess == difficulty['easy']:
			print("The number {0} is right :)".format(guess))
			tries = tries + 1
			if tries > 1:
				print("You've nedded {0} tries".format(tries))
			else:
				print("You've nedded {0} try".format(tries))
			exit()

		if guess > difficulty['easy']:
			print("The number {0} is too big".format(guess))
			tries = tries + 1

		if guess < difficulty['easy']:
			print("The number {0} is too small".format(guess))
			tries = tries + 1

if choice == "2":
	tries = 0
	while True:
		print("Enter your guess: ")
		g = input()
		guess = int(g)
		if guess == difficulty['medium']:
			print("The number {0} is right :)".format(guess))
			tries = tries + 1
			if tries > 1:
				print("You've nedded {0} tries".format(tries))
			else:
				print("You've nedded {0} try".format(tries))
			exit()

		if guess > difficulty['medium']:
			print("The number {0} is too big".format(guess))
			tries = tries + 1

		if guess < difficulty['medium']:
			print("The number {0} is too small".format(guess))
			tries = tries + 1

if choice == "3":
	tries = 0
	while True:
		print("Enter your guess: ")
		g = input()
		guess = int(g)
		if guess == difficulty['hard']:
			print("The number {0} is right :)".format(guess))
			tries = tries + 1
			if tries > 1:
				print("You've nedded {0} tries".format(tries))
			else:
				print("You've nedded {0} try".format(tries))
			exit()

		if guess > difficulty['hard']:
			print("The number {0} is too big".format(guess))
			tries = tries + 1

		if guess < difficulty['hard']:
			print("The number {0} is too small".format(guess))
			tries = tries + 1

if choice == "4":
	tries = 0
	while True:
		print("Enter your guess: ")
		g = input()
		guess = int(g)
		if guess == difficulty['very hard']:
			print("The number {0} is right :)".format(guess))
			tries = tries + 1
			if tries > 1:
				print("You've nedded {0} tries".format(tries))
			else:
				print("You've nedded {0} try".format(tries))
			exit()

		if guess > difficulty['very hard']:
			print("The number {0} is too big".format(guess))
			tries = tries + 1

		if guess < difficulty['very hard']:
			print("The number {0} is too small".format(guess))
			tries = tries + 1



		

File: GuessMyNumberGame.rb
puts "Hello and welcome to Roman Menzel's GuessMyNumberGame :)"
difficulty = {'easy' => rand(10), 'medium' => rand(100), 'hard' => rand(1000), 'very hard' => rand(10000)}
tries = 0

puts "Selcect a difficulty"
puts "1.) Easy"
puts "2.) Medium"
puts "3.) Hard"
puts "4.) Very hard"
print "Your choice: "
sel = gets.chomp.to_f

if sel == 1 then
	while true 
	print "Enter your guess: "
	number =  gets.chomp.to_f
	if number == difficulty['easy'] then
		puts "The number #{number} is right :)"
		tries += 1
		if tries > 1
			puts "You've needed #{tries} tries"
			exit
		else
			puts "You've needed #{tries} try"
		exit
		end
	end
	if number < difficulty['easy'] then
		puts "The number #{number} is too small"
		tries += 1
	end
	if number > difficulty['easy'] then
		puts "The number #{number} is too big"
		tries += 1
	end
	end
end

if sel == 2 then
	while true 
	print "Enter your guess: "
	number =  gets.chomp.to_f
	if number == difficulty['medium'] then
		puts "The number #{number} is right :)"
		tries += 1
		if tries > 1
			puts "You've needed #{tries} tries"
			exit
		else
			puts "You've needed #{tries} try"
		exit
		end
		exit
	end
	if number < difficulty['medium'] then
		puts "The number #{number} is too small"
		tries += 1
	end
	if number > difficulty['medium'] then
		puts "The number #{number} is too big"
		tries += 1
	end
	end
end

if sel == 3 then
	while true 
	print "Enter your guess: "
	number =  gets.chomp.to_f
	if number == difficulty['hard'] then
		puts "The number #{number} is right :)"
		tries += 1
		if tries > 1
			puts "You've needed #{tries} tries"
			exit
		else
			puts "You've needed #{tries} try"
		exit
		end
		exit
	end
	if number < difficulty['hard'] then
		puts "The number #{number} is too small"
		tries += 1
	end
	if number > difficulty['hard'] then
		puts "The number #{number} is too big"
		tries += 1
	end
	end
end

if sel == 4 then
	while true 
	print "Enter your guess: "
	number =  gets.chomp.to_f
	if number == difficulty['very hard'] then
		puts "The number #{number} is right :)"
		tries += 1
		if tries > 1
			puts "You've needed #{tries} tries"
			exit
		else
			puts "You've needed #{tries} try"
		exit
		end
		exit
	end
	if number < difficulty['very hard'] then
		puts "The number #{number} is too small"
		tries += 1
	end
	if number > difficulty['very hard'] then
		puts "The number #{number} is too big"
		tries += 1
	end
	end
end
